### DISCUSSION TITLE
How can I share values from response in next then methods?

### DISCUSSION THREAD

**Discussion by Lumca** [cypress-io/cypress#15471] [Questions and Help]
I am trying to write a custom command for my app where I create a real user and use its ID in set of tests.

Since I do test on real BE and DB and I do not stub or mock any data I need to store some values.

The usecase is:

> I send request
> I get response with unique ID
> I use this ID in another request and so on..

Here is example code I try to run
`cy.request('POST', Cypress.env('active') + 'api/players', {})
    .then(response => {
      expect(response.status).to.eq(200)
      const playerID = response.body.id
      return playerID
    })
    .then(playerID => {
      cy.request('POST', Cypress.env('active') + 'api/kyc/processes', {
        "level": "player",
        "player": playerID,
        "venue": 1
      })
    })
    .then(response => {
      expect(response.status).to.eq(200);
      const registrationID = response.body.id
      return registrationID
    })
    .then((playerID, registrationID) => {
      cy.request('POST', Cypress.env('active') + `api/players/${playerID}/attach`, {
        "data": "a",
        "mimeType": "image/png",
        "fileName": `front-id-card-player-${registrationID}.png`
      })
        .then((response) => {
          expect(response.status).to.eq(200);
        })
    })
    .then((playerID, registrationID) => {
      cy.request('PUT', Cypress.env('active') + `/api/players/${playerID}`, {
        "properties": {
          "type_of_id": "ID_CARD"
        }
      }).then((response) => {
        expect(response.status).to.eq(200);
      });
    });`

**Comment by Lumca** [comment]
So I have tried to writeFile readFile and it works.
But I think it should be way easier to do as it is just passing data trough few request.

**Comment by bahmutov** [comment]
Not exactly describing what you need but read this first https://on.cypress.io/network-requests and I would probably need to add an example there
